src/Core/Framework/DataAbstractionLayer/Write/WriteCommandExtractor.php

```diff
@@ -353,7 +355,11 @@ private function getMainFields(array $fields): array
                 continue;
             }
 
-            if ($field instanceof FkField) {
+            if (
+                $field instanceof FkField
```

Comment:
this case is already handled by the previous if condition. CreatedAt and UpdatedAt are not part of the primary key.

Reply:
When you look at `UpdatedAtFieldSerializer.php` the value will not be filled when the entity not yet exists (which happens before the actual insert). Of course the code would break here if the database field is required.

BUT: The test `testUpdatedAtWillBeSetAutomatically` have an issue. The `updated_at` field here is not required, but should be.

```php
<?php declare(strict_types=1);

namespace Shopware\Core\Framework\DataAbstractionLayer\FieldSerializer;

use Shopware\Core\Framework\DataAbstractionLayer\Exception\InvalidSerializerFieldException;
use Shopware\Core\Framework\DataAbstractionLayer\Field\Field;
use Shopware\Core\Framework\DataAbstractionLayer\Field\UpdatedAtField;
use Shopware\Core\Framework\DataAbstractionLayer\Write\DataStack\KeyValuePair;
use Shopware\Core\Framework\DataAbstractionLayer\Write\EntityExistence;
use Shopware\Core\Framework\DataAbstractionLayer\Write\WriteParameterBag;

class UpdatedAtFieldSerializer extends DateTimeFieldSerializer
{
    /**
     * @throws InvalidSerializerFieldException
     */
    public function encode(
        Field $field,
        EntityExistence $existence,
        KeyValuePair $data,
        WriteParameterBag $parameters
    ): \Generator {
        if (!$field instanceof UpdatedAtField) {
            throw new InvalidSerializerFieldException(UpdatedAtField::class, $field);
        }
        // Why so?
        if (!$existence->exists()) {
            return;
        }

        $data->setValue(new \DateTime());

        yield from parent::encode($field, $existence, $data, $parameters);
    }
}
```

Reply:
Push

Reply:
@OliverSkroblin push

Reply:
But this is not an issue. It is your iteration that this field is no longer defined as Required. Which is also correct, since it does not have to be set externally. That means the DAL takes care of it automatically. Therefore the test has to check exactly that.

Reply:
@OliverSkroblin I don't understand what you want to tell me. Could you explain this in german, please? :D

Reply:
I fixed the problems. Sorry for the late answer. For some reasons git did not close this merge request automatically.

https://github.com/shopware/platform/commit/e52dcf6b18a3c33f24a3b328546cf0c0be06dd39